Sources/HAP/Endpoints/characteristics().swift

```diff
@@ -89,15 +146,32 @@ func characteristics(device: Device) -> Application {
 
                 // toggle events for this characteristic on this connection
                 if let events = item["ev"] as? Bool {
+                    guard characteristic.permissions.contains(.events) else {
+                        serialized.append(["aid": aid,"iid": iid,"status": HAPStatusCodes.notificationNotSupported.rawValue])
+                        multiStatusResponse = true
+                        break
+                    }
                     if events {
                         device.add(characteristic: characteristic, listener: connection)
                         logger.info("Added listener for \(characteristic)")
                     } else {
                         device.remove(characteristic: characteristic, listener: connection)
                         logger.info("Removed listener for \(characteristic)")
                     }
+                    serialized.append(["aid": aid,"iid": iid,"status": HAPStatusCodes.success.rawValue])
                 }
             }
+            
+            if multiStatusResponse {
+                do {
+                    let json = try JSONSerialization.data(withJSONObject: ["characteristics": serialized], options: [])
+                    return Response(status: serialized.count == 1 ? .badRequest : .multiStatus, data: json, mimeType: "application/hap+json")
```

Comment:
Instead of setting `multiStatusResponse`, can't we keep an array of responses? And when it's time to create the response, see if there's a common denominator which we can return and `.multiStatus` otherwise? Is there a specific reason you introduced this state flag?

Reply:
See your point, flags are difficult to maintain.